src/Xablu.WebApiClient/Services/GraphQL/Request.cs

```diff
@@ -109,6 +97,34 @@ private void GetProperties(Type type)
             }
         }
 
+        private void SetDictionary(PropertyInfo property, Dictionary<string, object> propDict)
+        {
+            var hasAttribute = Attribute.IsDefined(property, typeof(QueryParameter)) || Attribute.IsDefined(property, typeof(QueryName));
+
+            if (hasAttribute)
+            {
+
+
+
+                var isExclusive = !string.IsNullOrEmpty((Attribute.GetCustomAttribute(property, typeof(QueryParameter)) as QueryParameter)?.ExclusiveWith);
+
+
+
+                if (!isExclusive)
+                {
+                    var queryName = (Attribute.GetCustomAttribute(property, typeof(QueryName)) as QueryName)?.Values[0];
+                    var query = !string.IsNullOrEmpty(queryName) ? $"{property.Name}: {queryName}" : $"{property.Name}" + $"{{{attributeNumber.ToString()}}}";
+
+                    propDict.Add(query, null);
+                    attributeNumber++;
+                }
+            }
+            else
+            {
+                propDict.Add(property.Name, null);
+            }
```

Comment:
Does this logic mean that `QueryParameter` and `QueryName` attributes cannot be used in combination? I would guess that's something one might want to do, as the attributes serve different purposes

Reply:
No because it checks if any of these two attributes are currently attached, if not then just add it regulary to the query without doing anything. See code below

`var hasAttribute = Attribute.IsDefined(property, typeof(QueryParameterAttribute)) || Attribute.IsDefined(property, typeof(NameOfItemAttribute));`

Reply:
so in other words they can be used together. It only checks if either one of them are currently attached.